### PR TITLE
Fix upgrade tests in release-1.1.

### PR DESCRIPTION
**Pull request by hklai**
Ported various fixes from release-1.0 and also
https://github.com/istio/istio/pull/9939.

### CODE REVIEW

File: tests/upgrade/test_crossgrade.sh [hunk 1]
```diff
@@ -116,38 +117,55 @@ installIstioSystemAtVersionHelmTemplate() {
         auth_opts="--set global.mtls.enabled=true --set global.controlPlaneSecurityEnabled=true "
     fi
     release_path="${3}"/install/kubernetes/helm/istio
-    helm init --client-only
-    helm dep update "${release_path}"
+    if [[ "${release_path}" != *"1.0"* ]]; then
```
Comment (ostromart) [inline tests/upgrade/test_crossgrade.sh]: I'd leave this as == 1.1 for one very good reason that you're pointing to 1.1 release below.

Comment (hklai) [reply tests/upgrade/test_crossgrade.sh]: OK you win, for now. =)

Will probably fix this by adding charts to the release tarball so we don't need to rely on that URL here.

**Pull request merged**